[UnwilledTangent] iP

UnwilledTangent opened the pull request.

Commit:
Add Gradle support

UnwilledTangent committed:
Completed Duke Level-1: Greet, Echo, Exit

UnwilledTangent committed:
Completed Level 2. Add, List

UnwilledTangent committed:
Completed Level 3. Mark as Done.

UnwilledTangent committed:
Complete Level 4: ToDos, Events, Deadlines

UnwilledTangent committed:
Done Automated Text UI Testing

UnwilledTangent committed:
Complete Level 5. Handle Errors

UnwilledTangent committed:
Completed Level 6. Delete

UnwilledTangent committed:
Changed error testing

UnwilledTangent committed:
Do Level-7 Save

UnwilledTangent committed:
Split into Java classes

UnwilledTangent committed:
Delete past static classes in Duke.java

UnwilledTangent committed:
Do Level-8 Dates and Times

UnwilledTangent committed:
Merge branch 'branch-Level-8'

UnwilledTangent committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

UnwilledTangent committed:
Do A-MoreOOP

UnwilledTangent committed:
Do A-Packages

UnwilledTangent committed:
Add JUnit test

UnwilledTangent committed:
Add JavaDoc comments

UnwilledTangent committed:
Do A-CodingStandard

UnwilledTangent committed:
Do Level-9 Find

UnwilledTangent committed:
Merge branch 'branch-A-CodingStandard'

UnwilledTangent committed:
Merge branch 'branch-Level-9'

UnwilledTangent committed:
Redo A-MoreOOP

UnwilledTangent committed:
Add miscellaneous files

StanleyNeoh commented on src/main/duke/src/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+ * Contains Deadlines, a subclass of Task
+ */
+public class Deadline extends Task {
+    private final String taskType = "[D]";
```
As this is final, maybe you might want to consider fully capitalising according to the coding standard. e.g. TASK_TYPE

StanleyNeoh commented on src/main/duke/src/Event.java:
```diff
@@ -0,0 +1,38 @@
+ * Class for Event, a subclass of Task
+ */
+public class Event extends Task{
+    private final String taskType = "[E]";
```
Same thing here about the naming style

StanleyNeoh commented on src/main/duke/src/Task.java:
```diff
@@ -0,0 +1,52 @@
+ * Super class of Deadline, Event, and ToDo
+ */
+public class Task {
+    protected  String description;
```
looks like there is 2 spaces between "protected" and "String"

Merrickneo replied:
Is there a particular reason why the access modifier is `protected` instead of `private`?

StanleyNeoh commented on src/main/duke/src/TaskList.java:
```diff
@@ -0,0 +1,106 @@
+    public void unmarkTask(String input) {
+        int taskNumber = Integer.parseInt(input.split(" ")[1]) - 1;
+        tasksList.get(taskNumber).markAsNotDone();
+        storage.saveTodoList(tasksList);
```
Seems that you named your Todo with a lowercase 'd' here but your function addToDo uses an uppercase 'D'. You may want to consider keeping it consistent throughout.

StanleyNeoh commented on src/main/duke/src/ToDo.java:
```diff
@@ -0,0 +1,24 @@
+ * subclass of Task
+ */
+public class ToDo extends Task {
+    private final String taskType = "[T]";
```
Same thing about the name styling here.

StanleyNeoh commented on src/main/java/duke/duke/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+ * Contains Deadlines, a subclass of Task
+ */
+public class Deadline extends Task{
+    private final String taskType = "[D]";
```
and here also.

StanleyNeoh commented on src/main/java/duke/duke/Duke.java:
```diff
@@ -0,0 +1,178 @@
+
+public class Duke {
+    /**
+     * main class for Duke todolist
```
May want to consider capitalising the first letter of main.

StanleyNeoh replied:
Calling it todolist might be misleading as it contains events, deadlines and todos. May want to consider updating the name.

StanleyNeoh commented on src/main/java/duke/duke/Event.java:
```diff
@@ -0,0 +1,40 @@
+ * Class for Event, a subclass of Task
+ */
+public class Event extends Task{
+    private final String taskType = "[E]";
```
Same thing here.

StanleyNeoh commented on src/main/java/duke/duke/Task.java:
```diff
@@ -0,0 +1,54 @@
+ * Super class of Deadline, Event, and ToDo
+ */
+public class Task {
+    protected  String description;
```
Looks like there is 2 spaces between protected and String here as well.

StanleyNeoh reviewed (commented):
Overall looks pretty good. Just some minor naming style violations. I also think you don't need to commit the binary files as they can be easily generated.

UnwilledTangent commented:
Thank you @StanleyNeoh , for the very very detailed review! I will edit the code as soon as I can!

Merrickneo commented on src/main/duke/src/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.byTime = LocalDate.parse(by);
```
Could have a more descriptive name for `LocalDate`  like perhaps `LocalDateParser` for easier interpretation of this static method.

Merrickneo commented on src/main/duke/src/Duke.java:
```diff
@@ -0,0 +1,13 @@
+    private static TaskList tasks = new TaskList();
+    private static Ui ui = new Ui();
+    private static Parser parser = new Parser();
+//    Scanner scanner = new Scanner(System.in);
```
Can remove if not required!

Merrickneo commented on src/main/duke/src/Event.java:
```diff
@@ -0,0 +1,38 @@
+     */
+    public Event(String description, String startTime, String endTime) {
+        super(description);
+
```
Could remove the space to be more compact.

Merrickneo commented on src/main/duke/src/Task.java:
```diff
@@ -0,0 +1,52 @@
+     */
+    public void markAsDone() {
+        this.isDone = true;
+        System.out.println("Nice! I've marked this task as done:");
```
Since the `Ui` class is supposed to handle the user input and interface, consider making use of `Ui` to do the printing instead!

Merrickneo commented on src/main/java/duke/duke/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    /**
+     *
```
Extra line here.

Merrickneo reviewed (commented):
Excellent Work here! Could see the effort you put in for your iP.

Commented on several minor details and suggestions you could take up if you want.
OOP principles are largely adhered to which is great! Keep it up!

UnwilledTangent commented:
Thank you so much @Merrickneo for being so thorough! I will update the code according to your recommendations asap!

UnwilledTangent committed:
Change build.gradle file

UnwilledTangent committed:
Refactor to Java file

UnwilledTangent committed:
Change build.gradle for Java FX

UnwilledTangent committed:
Change duke.duke to duke

UnwilledTangent committed:
Refactor to different package for file management

UnwilledTangent committed:
Delete production files

UnwilledTangent committed:
Do JavaFX Tutorial Part 1

UnwilledTangent committed:
Before changing parser to fit into getResponse()

UnwilledTangent committed:
Do JavaFX tutorial part 3, Iteration 2 – Adding Dialog Boxes

UnwilledTangent committed:
Do ALL JavaFX Tutorial Part 3

UnwilledTangent committed:
Do finish Level-10

UnwilledTangent committed:
Delete dead code / comments

UnwilledTangent committed:
Delete if-else statement for newFile.createNewFile()

UnwilledTangent committed:
Removed dead code

Description: Dead code is bad. Anyway, this is just a sample commit message body to fulfill A-FullCommitMessage

UnwilledTangent committed:
Add assertion

Assertions are important to reduce bugs in programs

UnwilledTangent committed:
Improve Code Quality

Many thanks to SoC SE Script\nMerrcik Neo @Merrickneo\nStanley Neoh @StanleyNeoh

UnwilledTangent committed:
Add help page

UnwilledTangent committed:
Add A-UserGuide

UnwilledTangent committed:
Edit build.gradle main class path to Launcher.java

UnwilledTangent committed:
Edit read.md file

UnwilledTangent committed:
Edit read.md file

UnwilledTangent committed:
Add javadocs comments
